### PR TITLE
Easy Testing

### PR DESCRIPTION
**Pull request by Nusrik**
- Added function for reading all files from a directory and comparing with the golden files
- RunTest function optimization
- Function optimization

### CODE REVIEW

File: Caches/test/EteTests/testRunner.py [hunk 1]
```diff
@@ -14,9 +14,9 @@ def isExe(fpath):
         return ret
 
 def runTest(execFile, testPath):
-    inputFile = open(testPath, 'r')
-    data = "".join(inputFile.readlines())
-    inputFile.close() 
+    """Run the test and return the output."""
```
Comment (Jerma066) [inline Caches/test/EteTests/testRunner.py:17]: Maybe it is better to move this comment before the function defenition?
(This comment describes the whole function)

File: Caches/test/EteTests/testRunner.py [hunk 2]
```diff
@@ -29,16 +29,13 @@ def findDiff(first, second):
     return chr(c)
  
 def chechTestResult(goldenPath, testResult):
-    goldenFile = open(goldenPath, 'r')
-    expectedResult = "".join(goldenFile.readlines())
-    goldenFile.close()
-    checkStatus = "FAIL"
+    """Check the test result against the golden file."""
```
Comment (Jerma066) [inline Caches/test/EteTests/testRunner.py:32]: Same as previous:
>> Maybe it is better to move this comment before the function defenition?

File: Caches/test/EteTests/testRunner.py [hunk 3]
```diff
@@ -29,16 +29,13 @@ def findDiff(first, second):
     return chr(c)
  
 def chechTestResult(goldenPath, testResult):
```
Comment (Jerma066) [inline Caches/test/EteTests/testRunner.py]: Could you please also fix misprint in the function name: chechTestResult -> checkTestResult?

File: Caches/test/EteTests/testRunner.py (hunk 1 above)
Comment (Nusrik) [reply Caches/test/EteTests/testRunner.py:17]: such a description allows you to see a description of the function when you hover the cursor or request its description

File: Caches/test/EteTests/testRunner.py [hunk 4]
```diff
@@ -62,81 +59,15 @@ def chechTestResult(goldenPath, testResult):
     print("Executable main file: % s" % args.executable)
     print("Directory containing tests: % s" % args.test_dir)
 
-    testResult = runTest(args.executable, args.test_dir + "test1.dat")
-    checkResult = chechTestResult(args.test_dir + "golden/test1.ans.dat", testResult)
-    print(checkResult)
+    test_files = [f for f in os.listdir(args.test_dir) if f.endswith('.dat') and not f.startswith('golden')]
```
Comment (Jerma066) [inline Caches/test/EteTests/testRunner.py]: All variables are named using CamelCase.
Could you please raname test_files variable (or raname all other variables in one non-CamelCase style)?

File: Caches/test/EteTests/testRunner.py [hunk 5]
```diff
@@ -62,81 +59,15 @@ def chechTestResult(goldenPath, testResult):
...
 
-    testResult = runTest(args.executable, args.test_dir + "test2.dat")
-    checkResult = chechTestResult(args.test_dir + "golden/test2.ans.dat", testResult)
-    print(checkResult)
+    for test_file in test_files:
```
Comment (Jerma066) [inline Caches/test/EteTests/testRunner.py]: test_path and test_files and golden_path  also better to rename using CamelCase or get rid of CamelCase in whole program.

File: Caches/test/EteTests/testRunner.py (hunk 4 above)
Comment (Jerma066) [inline Caches/test/EteTests/testRunner.py]: Can you please support a recursive traversal of all nested directories with possible tests?

**Pull request merged**